src/LSM6DS3.cpp

```diff
@@ -176,14 +178,43 @@ float LSM6DS3Class::gyroscopeSampleRate()
   return 104.0F;
 }
 
+int LSM6DS3Class::readTemperature(float& t)
+{
+  int16_t data[1];
+
+  if (!readRegisters(LSM6DS3_OUT_TEMP_L, (uint8_t*)data, sizeof(data))) {
+    t = NAN;
+
+    return 0;
+  }
+
+  t = data[0] / 16 + 25;
```

Comment:
```suggestion
  t = data[0] / 16.0 + 25;
```
The previous integer math caused the temperature data to be truncated.

Reply:
Good catch!